### ISSUE TITLE
multistore currency conversions issues in catalog/controller/payment/pp_express.php?

### ISSUE THREAD

**Issue by pine3ree** [opencart/opencart#2601]
Please verify!

the code
```php
$max_amount = $this->currency->convert($this->cart->getTotal(), $this->config->get('config_currency'), 'USD');
```
is supposed to convert to USD currency value the cart total amount.

The cart total is given in the overall default currency which is the one used in all calculations and product prices in the administration area and which is the config_currency of the default store (store_id=0). But every store can have a different config_currency.

example:
cart total = 100 EUR (EUR being the default currency of store 0)

suppose we are in store 1 with config_currency = GBP

GBP currency value = 0.76
USD curency value = 1.14

the cart total in USD should be 100 * 1.14 = 114

while using the convert function we would have:

convert formula: value * (to_value /  from_value)

100 * (1.14 / 0.76)  = 150

in short

the cart total currency unit is not always equal to config_currency, it's only always equal to the config_currency of store 0. Conversions should take that into account.

**Comment by jamesallsup** [comment]
The code mentioned doesn't have any effect on the cart total, it is only used to calculate a higher value to be authenticated through the PayPal flow.

If you have a multi store set up and have issues with totals not matching then please update with some screen shots and live examples.

**Comment by pine3ree** [comment]
what i meant is different thing:

the code convert the cart total from config_currency of the current to USD.
the cart total is always in the config_currency of store 0, but config_currency can be different in different store.

what i am saying is that if prices are saved in EUR in the db,  $this->cart->getTotal() will give you result in EUR anywhere.

So what's the point in converting from for instance from GBP to USD a value that is calculated in EUR?
if store 1 has config_currency set to GBP that code will do exactly that.
